### PR TITLE
Model well known metadata, well known metadata keys are case insensitive

### PR DESCRIPTION
**Pull request by yufeih**
This turns `Robots` --> `robots`. #3434

### CODE REVIEW

File: src/docfx/build/legacy/metadata/LegacyMetadata.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ public static JObject GenerateLegacyRedirectionRawMetadata(Docset docset, PageMo
                 LegacyManifestOutput legacyManifestOutput,
                 TableOfContentsMap tocMap)
         {
-            var rawMetadata = pageModel.Metadata != null ? new JObject(pageModel.Metadata) : new JObject();
+            var rawMetadata = pageModel.Metadata != null ? JObject.FromObject(pageModel.Metadata) : new JObject();
```
Comment (928PJY) [inline src/docfx/build/legacy/metadata/LegacyMetadata.cs]: >JObject.FromObject( [](start = 59, length = 19)

Q: what is the difference?

Comment (yufeih) [reply src/docfx/build/legacy/metadata/LegacyMetadata.cs]: `FromObject` ensures the object is treated as json serialization. `JsonExtensionData` will be populated correctly to the result

**Comment by live1206** [comment]
>             var title = markup.Metadata.Value<string>("title") ?? HtmlUtility.GetInnerText(htmlTitleDom);

retrieve from property

---
Refers to: src/docfx/build/page/BuildPage.cs:99 in ae85dc3. [](commit_id = ae85dc31e5efe6ed814f47d9517a7e49ca5c18af, deletion_comment = False)

**Comment by yufeih** [comment]
>             var title = markup.Metadata.Value<string>("title") ?? HtmlUtility.GetInnerText(htmlTitleDom);

title logic is a little bit more complex, I'll handle that later. There is a TODO below at SDP title handling

---
In reply to: [423033152](https://github.com/dotnet/docfx/pull/3438#issuecomment-423033152) [](ancestors = 423033152)

---
Refers to: src/docfx/build/page/BuildPage.cs:99 in ae85dc3. [](commit_id = ae85dc31e5efe6ed814f47d9517a7e49ca5c18af, deletion_comment = False)

**Pull request merged**